jerry-core/ecma/operations/ecma-regexp-object.c

```diff
@@ -376,12 +383,18 @@ re_match_regexp (re_matcher_ctx_t *re_ctx_p, /**< RegExp matcher context */
       {
         JERRY_TRACE_MSG ("Execute RE_OP_MATCH: match\n");
         *out_str_p = str_curr_p;
+#ifdef REGEXP_RECURSION_LIMIT
+        --re_ctx_p->recursion_depth;
+#endif /* REGEXP_RECURSION_LIMIT */
         return ECMA_VALUE_TRUE; /* match */
       }
       case RE_OP_CHAR:
       {
         if (str_curr_p >= re_ctx_p->input_end_p)
         {
+#ifdef REGEXP_RECURSION_LIMIT
+          --re_ctx_p->recursion_depth;
+#endif /* REGEXP_RECURSION_LIMIT */
```

Comment:
I am thinking this could be moved to a macro or an always inline function, reducing these changes to 1 line. What do you think?

Reply:
It is a good idea to exchange those three lines with a macro.